### DISCUSSION TITLE
dynamically added widgets

### DISCUSSION THREAD

**Discussion by itinerishu** [beeware/toga#1281] [Q&A]
Is there a way to dynamically add a widget? For example I want to add a table widget only after I read in and processed a file  to display the data in the table. Since I can't initialize an empty table(without headings), i want to add it dynamically.

**Comment by freakboy3742** [comment]
There's no requirement that you call all your widget creation/layout commands in the `startup()` method - you can invoke them in callbacks as well. There are several example projects that show this behaviour - the `layout` demo adds and removes images; the `table` demo adds and removes a column from a table.

**Reply by itinerishu** [reply]
Thank you Russel Keith-Magee!

I tested the examples and turned out there is a bug somewhere with the containers and their content.  I copied the layout example and when i ran it the image(i changed the path to the image)could not be shown in the scroll_view on the right. Pressing the reparent button works only when it is changing the parent to the simple box widget. However when the box widget is in the container so when the container's(scroll_view) content is the box widget and I try to add the image in this box the image won't show. I tried putting a simple label in that scroll_view to see if the problem is that it is empty initially but it looks like something is got added because the label changes position but the image is still not shown. Then i simply replaced the scroll_view with a box widget and it worked.  I am using windows 10.

**Reply by freakboy3742** [reply]
Thanks for the report - looks like you've found a bug. That's not unheard of - Toga is at v0.3, and the Windows backend is the least developed of the three desktop platforms.

In this case, the issue relates to resizing widgets. We haven't worked out why (yet!), but on Windows, widgets aren't resized when their content changes; so the *initial* size for the widget is what sticks. This is a problem when you start adding/removing content, as the the layout *should* resize itself, but it doesn't.

I thought there was an open ticket for this, but I haven't been able to find one; I've opened #1284 to track the problem.

**Reply by itinerishu** [reply]
It might also be related.? The window positioning(place on screen) does not seem to work either.
